wasm-host/src/host_functions.rs

```diff
@@ -480,3 +482,128 @@ pub fn oracle_keylet(
     set_data(keylet_hash.len() as i32, out_buf_ptr, keylet_hash, _caller)?;
     Ok(vec![WasmValue::from_i32(HASH256_LEN as i32)])
 }
+
+pub fn trace(
+    _data_provider: &mut DataProvider,
+    _inst: &mut Instance,
+    _caller: &mut CallingFrame,
+    inputs: Vec<WasmValue>,
+) -> Result<Vec<WasmValue>, CoreError> {
+    // Expect 5 inputs.
+
+    // check the number of inputs
+    if inputs.len() != 5 {
```

Comment:
Consider this form, or also helper functions to ensure valid inputs.
https://github.com/ripple/craft/pull/19/files#diff-a8acdb4be5b4320567bcb977ad4f80a9587af152bc88d92017369b142772ec74R181